orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -1714,6 +1714,26 @@ IMPLEMENT_GRADIENT_BUILDER(GetExternalGradient) {
   return result;
 }
 
+IMPLEMENT_GRADIENT_BUILDER(GetATenOpGradient) {
+  size_t num_inputs = GetSrcNodeInputSize();
+  std::vector<int64_t> requires_grad;
+  std::vector<ArgDef> output_args;
+  for (size_t i = 0; i < num_inputs; i++) {
+    if (IsGradientRequiredForSrcNodeInput(i)) {
```

Comment:
I realized that we are missing a mechanism to describe the differentiability for generic ATenOp. For regular ops, we use STOP_GRADIENT_EDGES to indicate which input is not differentiable. 

For ATenOp, we may introduce an attribute called "differentiable_inputs" to indicate which input is differentiable. 

Notice that this is different from the "requires_grad" attribute you are adding.